test/GatewayUpgradeFlow.js

```diff
@@ -0,0 +1,180 @@
+'use strict';
+
+const chai = require('chai');
+const { sortBy } = require('lodash');
+const { ethers, network } = require('hardhat');
+const {
+    utils: { defaultAbiCoder, Interface, solidityKeccak256 },
+} = ethers;
+const { expect } = chai;
+const { isHardhat, getAddresses, getWeightedAuthDeployParam, getWeightedProxyDeployParams } = require('./utils');
+const { getBytecodeHash } = require('@axelar-network/axelar-contract-deployments');
+
+const getWeights = ({ length }, weight = 1) => Array(length).fill(weight);
+
+describe('InterchainGovernance', () => {
+    const threshold = isHardhat ? 4 : 2;
+
+    let ownerWallet;
+    let governanceAddress;
+
+    let interchainGovernanceFactory;
+    let interchainGovernance;
+    let gatewayFactory;
+    let authFactory;
+    let gatewayProxyFactory;
+    let tokenDeployerFactory;
+
+    let auth;
+    let tokenDeployer;
+    let gateway;
+
+    let wallets;
+    let owner;
+    let operators;
+    let mintLimiter;
+
+    const governanceChain = 'Governance Chain';
+
+    before(async () => {
+        wallets = await ethers.getSigners();
+        ownerWallet = wallets[0];
+        mintLimiter = ownerWallet;
+        governanceAddress = wallets[1].address;
+        operators = sortBy(wallets.slice(0, threshold), (wallet) => wallet.address.toLowerCase());
+
+        interchainGovernanceFactory = await ethers.getContractFactory('TestInterchainGovernance', ownerWallet);
+
+        gatewayFactory = await ethers.getContractFactory('AxelarGateway', owner);
+        authFactory = await ethers.getContractFactory('AxelarAuthWeighted', owner);
+        gatewayProxyFactory = await ethers.getContractFactory('AxelarGatewayProxy', owner);
+        tokenDeployerFactory = await ethers.getContractFactory('TokenDeployer', owner);
+        tokenDeployer = await tokenDeployerFactory.deploy();
+        await tokenDeployer.deployTransaction.wait(network.config.confirmations);
+    });
+
+    beforeEach(async () => {
+        const buffer = 10 * 60 * 60;
```

Comment:
set it to 6s?

Reply:
made it similar to InterchainGovernance tests where buffer is dependent on whether the tests are running on hardhat or a live testnet